.github/workflows/ci.yml

```diff
@@ -79,26 +64,46 @@ jobs:
           target: riscv32imc-unknown-none-elf,riscv32imac-unknown-none-elf
           toolchain: stable
           components: rust-src
-      # Install the Rust nightly toolchain for RISC-V devices, because of `-Zdoctest-xcompile`
+      # Install the Rust nightly toolchain for RISC-V devices. TODO: with 1.89 doctest-xcompile became stable
       - uses: dtolnay/rust-toolchain@v1
         with:
           target: riscv32imc-unknown-none-elf,riscv32imac-unknown-none-elf
           toolchain: nightly
           components: rust-src
 
-      - uses: Swatinem/rust-cache@v2
-        with:
-          prefix-key: "ci-${{ matrix.device.soc }}"
-          cache-all-crates: true
+      - name: Setup cargo-batch
+        run: |
+          # Note the prebuilt is linux-only, the macOS runner should have cargo batch installed, but it needs to build it itself.
+          if ! command -v cargo-batch &> /dev/null; then
+              # mkdir -p $HOME/.cargo/bin
+              # curl -L https://github.com/embassy-rs/cargo-batch/releases/download/batch-0.6.0/cargo-batch > $HOME/.cargo/bin/cargo-batch
+              # chmod +x $HOME/.cargo/bin/cargo-batch
+              cargo install --git https://github.com/embassy-rs/cargo-batch cargo --bin cargo-batch --locked --force
+          fi
 
       - name: Build and Check
         shell: bash
-        run: cargo xcheck ci ${{ matrix.device.soc }} --toolchain ${{ matrix.device.toolchain }}
+        run: |
+          # lints and docs are checked separately
+          cargo xcheck ci esp32 --toolchain esp --no-lint --no-docs
```

Comment:
I'm a bit confused why we wouldn't want to lint here? Am I missing the linting happening elsewhere?

Reply:
Nightly CI runs the full test suite, msrv checks run the lint-packages command.